test/lit/passes/local-subtyping.wast

```diff
@@ -568,4 +571,48 @@
       (local.get $x)
     )
   )
+
+  ;; CHECK:      (func $try-table-catch-result (type $0)
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (block $catch (result anyref)
+  ;; CHECK-NEXT:    (try_table (catch $e-anyref $catch)
+  ;; CHECK-NEXT:     (nop)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (ref.null none)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $try-table-catch-result
+    (drop
+      ;; Must not be refined to nullref
+      (block $catch (result anyref)
+        (try_table (catch $e-anyref $catch)
+          (nop)
+        )
+        (ref.null none)
+      )
+    )
+  )
+
+  ;; CHECK:      (func $try-table-ref (type $0)
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (block $catch (result exnref)
+  ;; CHECK-NEXT:    (try_table (catch_all_ref $catch)
+  ;; CHECK-NEXT:     (nop)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (ref.null noexn)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $try-table-ref
+    (drop
+      ;; Must not be refined to nullexnref
+      (block $catch (result exnref)
+        (try_table (catch_all_ref $catch)
+          (nop)
+        )
+        (ref.null exn)
+      )
+    )
+  )
```

Comment:
Is this test needed? It seems very similar to the one above it.

Reply:
One tests the case where the value is sent because it's part of the exception's payload. The other is when it's the `_ref` that generates an `exnref`. It's actually the latter I ran into in a real context.

To me those are distinct causes, which deserve distinct tests. But if you disagree, I'm happy to remove one of them.

Reply:
Ah, good point. Yes, those are distinct causes, let's keep the new test.